### PR TITLE
Managed vault - add share price to VaultInfo query

### PR DESCRIPTION
**Pull request by piobab**

### CODE REVIEW

File: contracts/vault/src/msg.rs [hunk 1]
```diff
@@ -98,6 +98,15 @@ pub struct VaultInfoResponseExt {
 
     /// Performance fee configuration
     pub performance_fee_config: PerformanceFeeConfig,
+
+    /// Total base tokens in the vault
+    pub total_base_tokens: Uint128,
```
Comment (piobab) [inline contracts/vault/src/msg.rs:103]: @bobthebuidlr @Subham2804 do you think it is good place for these data?

Comment (Subham2804) [reply contracts/vault/src/msg.rs:103]: Looks good to me (will require no other query for backend API implementation)

**Review by Subham2804 — APPROVED** [review]

File: contracts/vault/src/msg.rs (hunk 1 above)
Comment (Subham2804) [reply contracts/vault/src/msg.rs:103]: looks good to me

Comment (piobab) [reply contracts/vault/src/msg.rs:103]: @Subham2804 do you index the rest from this query? I mean title etc

Comment (Subham2804) [reply contracts/vault/src/msg.rs:103]: we do use subtitles , vault name etc from this query

Comment (bobthebuidlr) [reply contracts/vault/src/msg.rs:103]: Yes this looks like a fine place to me too

File: contracts/vault/src/query.rs [hunk 2]
```diff
@@ -14,16 +14,35 @@ use crate::{
...
+    // It means that the vault is not binded to any account yet.
+    let vault_account_id_opt = VAULT_ACC_ID.may_load(deps.storage)?;
+    let mut total_base_tokens = Uint128::zero();
+    let mut share_price = None;
+    if vault_account_id_opt.is_some() {
+        total_base_tokens = total_base_tokens_in_account(deps)?;
+        share_price = if total_vault_tokens.is_zero() {
+            None
+        } else {
+            Some(Decimal::checked_from_ratio(total_base_tokens, total_vault_tokens)?)
```
Comment (bobthebuidlr) [inline contracts/vault/src/query.rs:30]: Maybe we need to place a comment that the share_price is always denominated in the base token denom, and not usdc per se.

**Review by bobthebuidlr — APPROVED** [review]

File: contracts/vault/src/query.rs (hunk 2 above)
Comment (piobab) [reply contracts/vault/src/query.rs:30]: @bobthebuidlr please take a look on latest commit.

```
    /// Current share price, representing the value of one vault token in terms of the base token.
    /// It is calculated as the ratio of total base tokens in the vault to the total supply of vault tokens.
    /// Denominated as a decimal value:
    /// `share_price = total_base_tokens / total_vault_tokens`
    /// This share price allows users to determine how many base tokens can be redeemed per vault token:
    /// `base_tokens = vault_share * vault_tokens`
    pub share_price: Option<Decimal>,
```

Comment (bobthebuidlr) [reply contracts/vault/src/query.rs:30]: Beautiful!

**Pull request merged**